clinica/pipelines/deeplearning_prepare_data/deeplearning_prepare_data_utils.py

```diff
@@ -0,0 +1,228 @@
+            extracted_slice_rgb_axi = torch.stack((slice_select_axi, slice_select_axi, slice_select_axi))  # shape should be 3 * W * L
+
+            # save into .pt format
+            if slice_mode == 'original':
+                output_file_original.append(
+                        os.path.join(
+                            basedir,
+                            os.path.basename(preprocessed_T1).split('.pt')[0]
+                            + '_axis-axi_originalslice-'
+                            + str(index_slice)
+                            + '.pt'
+                            )
+                        )
+                torch.save(extracted_slice_original_axi.clone(), output_file_original[index_slice_list])
+            elif slice_mode == 'rgb':
+                output_file_rgb.append(
+                        os.path.join(
+                            basedir,
+                            os.path.basename(preprocessed_T1).split('.pt')[0]
+                            + '_axis-axi_rgbslice-'
+                            + str(index_slice)
+                            + '.pt'
+                            )
+                        )
+                torch.save(extracted_slice_rgb_axi.clone(), output_file_rgb[index_slice_list])
+
+    return output_file_rgb, output_file_original
+
+
+def extract_patches(preprocessed_T1, patch_size, stride_size):
+    """
+    This function extracts the patches from three directions
+    :param preprocessed_T1:
+    :return:
+    """
+    import torch
+    import os
+
+    basedir = os.getcwd()
+    image_tensor = torch.load(preprocessed_T1)
+
+    # use classifiers tensor.upfold to crop the patch.
+    patches_tensor = image_tensor.unfold(1, patch_size, stride_size).unfold(2, patch_size, stride_size).unfold(3, patch_size, stride_size).contiguous()
+    # the dimension of patch_tensor should be [1, patch_num1, patch_num2, patch_num3, patch_size1, patch_size2, patch_size3]
+    patches_tensor = patches_tensor.view(-1, patch_size, patch_size, patch_size)
+
+    output_patch = []
+    for index_patch in range(patches_tensor.shape[0]):
+        extracted_patch = patches_tensor[index_patch, ...].unsqueeze_(0)  # add one dimension
+        # save into .pt format
+        output_patch.append(
+                os.path.join(
+                    basedir,
+                    os.path.basename(preprocessed_T1).split('.pt')[0]
+                    + '_patchsize-'
+                    + str(patch_size)
+                    + '_stride-'
+                    + str(stride_size)
+                    + '_patch-'
+                    + str(index_patch)
+                    + '.pt'
+                    )
+                )
+        torch.save(extracted_patch.clone(), output_patch[index_patch])
+
+    return output_patch
+
+
+def save_as_pt(input_img):
+    """
+    This function transforms  nii.gz file into .pt format, in order to train
```

Comment:
To clarify: `input_img` can be `.nii` or `.nii.gz`, right?

Reply:
Yes, it uses as backend the load method available from nibabel (it reads both formats).